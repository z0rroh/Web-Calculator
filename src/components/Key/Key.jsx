import React from 'react';
import PropTypes from 'prop-types';
import './Key.css'

const Key = ({ keyType, keyValue, keyAction}) => {

   return(
      <div 
         onClick={() => keyAction(keyValue)}
         className={`key-container ${keyType}`}>
         <p className="key-value">
            { keyValue }
         </p>
      </div>
   )
}

Key.propTypes = {
   keyAction: PropTypes.func.isRequired,
   keyType: PropTypes.string.isRequired,
   keyValue: PropTypes.string.isRequired
}

export default Key