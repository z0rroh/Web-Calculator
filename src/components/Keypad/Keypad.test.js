import React from 'react';
import {Â shallow, mount } from 'enzyme'
import Keypad from './Keypad'
import Key from '../Key/Key'

describe('<Keypad/>', () => {
   let wrapp

   beforeEach(() => {
      wrapp = shallow(
         <Keypad
            callOperator={jest.fn()}
            numbers={[]}
            operators={[]}
            setOperator={jest.fn()}
            updateDisplay={jest.fn()}/>
      )
   });

   it('Should render correctly', () => {
      expect(wrapp).toMatchSnapshot()
   })

   it('Should render 4 div', () => {
      expect(wrapp.find('div').length).toEqual(4)
   })

   it('Should render a instance of Key component for each index of numbers, operators and the submit Key', () => {
      const numbers = ['0','1']
      const operators = ['+','-']
      const submit = 1
      const keyTotal = numbers.length + operators.length + submit
      wrapp.setProps({ numbers, operators })
      expect(wrapp.find('Key').length).toEqual(keyTotal)
   })

});

describe('<Keypad/> Mounted in DOM', () => {
   let wrapp;
   beforeEach(() => {
     wrapp = mount(
       <Keypad
         callOperator={jest.fn()}
         numbers={[]}
         operators={[]}
         setOperator={jest.fn()}
         updateDisplay={jest.fn()}
       />
     );
   });
 
   it('renders the values of numbers to the DOM', () => {
     wrapp.setProps({ numbers: ['0', '1', '2'] })
     expect(wrapp.find('.numbers-container').text()).toEqual('012');
   });
 
   it('renders the values of operators to the DOM', () => {
     wrapp.setProps({ operators: ['+', '-', '*', '/'] });
     expect(wrapp.find('.operators-container').text()).toEqual('+-*/');
   });
 });