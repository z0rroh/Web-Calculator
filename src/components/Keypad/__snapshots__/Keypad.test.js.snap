// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Keypad/> Should render correctly 1`] = `
<div
  className="keypad-container"
>
  <div
    className="numbers-container"
  />
  <div
    className="operators-container"
  />
  <div
    className="submit-container"
  >
    <Key
      keyAction={[Function]}
      keyType="submit-key"
      keyValue="="
    />
  </div>
</div>
`;
